### ISSUE TITLE
🐛 Bug Report: Unable to bind debugger breakpoints to internals in CLI generated projects in VSCode

### ISSUE THREAD

**Issue by hotspoons** [backstage/backstage#21690]
### 📜 Description

I was attempting to troubleshoot my GitLab catalog configuration, and no matter what I did, I could not get breakpoints to bind in VSCode. 

### 👍 Expected behavior

I should be able to set breakpoints and step through node modules like I can with every other js/ts project I've ever worked on.

### 👎 Actual Behavior with Screenshots

Any attempt to set a breakpoint in the source mapped generated JS under the `@backstage/plugin-catalog-backend-module-gitlab` node module would result in white circles instead of breakpoints:

![image](https://github.com/backstage/backstage/assets/851973/0f745fe3-f853-43be-87fb-a89203c8e5b7)

if I set a breakpoint in my project's code and start navigating into methods with the debugger:

![image](https://github.com/backstage/backstage/assets/851973/d3efb99b-5485-4e88-b8c0-6647b50d4683)

I end up in what looks like the webpack contents with no source map:

![image](https://github.com/backstage/backstage/assets/851973/dae7ed2b-bb93-4462-9640-62741f1f50f8)

I tried fiddling around with the internals of the webpack configuration by hacking the backstage cli launcher to no avail. 

I *am* able to set breakpoints if I clone the main backstage repo and run the big-ol project, but I can't do this with a generated project. 

### 👟 Reproduction steps

1. Generate a new project using the CLI, open in VSCode with the following VSCode launch config:
```json
{
            "type": "node",
            "request": "launch",
            "name": "Start FE and backend at the same time",
            "runtimeExecutable": "yarn",
            "runtimeArgs": ["dev"],
            "skipFiles": [
                "<node_internals>/**"
            ],
            "sourceMaps": true,
            "console": "internalConsole", 
            "outputCapture": "std",
            "program": "${file}",
            "envFile": "${workspaceFolder}/.env",
            "env": {
                "LOG_LEVEL": "debug"
            }
        }
```
2. Try setting a breakpoint in any code under `node_modules/@backstage` you anticipate will be executed when starting or running backstage
3. Observe the inability to set the breakpoint

### 📃 Provide the context for the Bug.

Tried with freshly generated backstage project, on two separate installs of VSCode

### 🖥️ Your Environment

VSCode w/ remote dev tools, connecting to Rocky 8 Linux machine from Windows 11 workstation

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by tudi2d** [comment]
Thanks for raising this! We actually just updated the documentation on this. Can you double check if that helps: https://backstage.io/docs/local-dev/debugging/?

**Comment by jamime** [comment]
I have the same issue. @tudi2d, the updated launch.json makes no difference.

I'm using a generated project. 

```
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Start Backend",
      "type": "node",
      "request": "launch",
      "args": ["package", "start"],
      "cwd": "${workspaceFolder}/packages/backend",
      "program": "${workspaceFolder}/node_modules/.bin/backstage-cli",
      "skipFiles": ["<node_internals>/**"],
      "console": "integratedTerminal",
    }
  ]
}

```

```
OS:   Darwin 22.6.0 - darwin/arm64
node: v18.12.0
yarn: 1.22.19
cli:  0.24.0 (installed)
backstage:  1.20.0

Dependencies:
  @backstage/app-defaults                                          1.4.6
  @backstage/backend-app-api                                       0.5.9
  @backstage/backend-common                                        0.19.9, 0.20.0
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.1.1
  @backstage/backend-plugin-api                                    0.6.8
  @backstage/backend-tasks                                         0.5.13
  @backstage/catalog-client                                        1.5.1
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.1
  @backstage/cli                                                   0.24.0
  @backstage/config-loader                                         1.6.0
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.2
  @backstage/core-compat-api                                       0.1.0
  @backstage/core-components                                       0.13.9
  @backstage/core-plugin-api                                       1.8.1
  @backstage/e2e-test-utils                                        0.1.0
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.4
  @backstage/frontend-plugin-api                                   0.4.0
  @backstage/integration-aws-node                                  0.1.8
  @backstage/integration-react                                     1.1.22
  @backstage/integration                                           1.8.0
  @backstage/plugin-api-docs                                       0.10.2
  @backstage/plugin-app-backend                                    0.3.56
  @backstage/plugin-app-node                                       0.1.8
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.0
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.2
  @backstage/plugin-auth-backend-module-github-provider            0.1.5
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.5
  @backstage/plugin-auth-backend-module-google-provider            0.1.5
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.5
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.0
  @backstage/plugin-auth-backend-module-okta-provider              0.0.1
  @backstage/plugin-auth-backend                                   0.20.2
  @backstage/plugin-auth-node                                      0.4.2
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.5
  @backstage/plugin-catalog-backend                                1.16.0
  @backstage/plugin-catalog-common                                 1.0.19
  @backstage/plugin-catalog-graph                                  0.3.2
  @backstage/plugin-catalog-import                                 0.10.4
  @backstage/plugin-catalog-node                                   1.6.0
  @backstage/plugin-catalog-react                                  1.9.2
  @backstage/plugin-catalog                                        1.16.0
  @backstage/plugin-events-node                                    0.2.17
  @backstage/plugin-github-actions                                 0.6.9
  @backstage/plugin-org                                            0.6.18
  @backstage/plugin-permission-common                              0.7.11
  @backstage/plugin-permission-node                                0.7.19
  @backstage/plugin-permission-react                               0.4.18
  @backstage/plugin-proxy-backend                                  0.4.6
  @backstage/plugin-scaffolder-backend-module-azure                0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.1.0
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.0
  @backstage/plugin-scaffolder-backend-module-github               0.1.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.11
  @backstage/plugin-scaffolder-backend                             1.19.2
  @backstage/plugin-scaffolder-common                              1.4.4
  @backstage/plugin-scaffolder-node                                0.2.9
  @backstage/plugin-scaffolder-react                               1.7.0
  @backstage/plugin-scaffolder                                     1.17.0
  @backstage/plugin-search-backend-module-catalog                  0.1.12
  @backstage/plugin-search-backend-module-pg                       0.5.17
  @backstage/plugin-search-backend-module-techdocs                 0.1.12
  @backstage/plugin-search-backend-node                            1.2.12
  @backstage/plugin-search-backend                                 1.4.8
  @backstage/plugin-search-common                                  1.2.9
  @backstage/plugin-search-react                                   1.7.4
  @backstage/plugin-search                                         1.4.4
  @backstage/plugin-tech-insights-backend-module-jsonfc            0.1.40
  @backstage/plugin-tech-insights-backend                          0.5.22
  @backstage/plugin-tech-insights-common                           0.2.12
  @backstage/plugin-tech-insights-node                             0.4.14
  @backstage/plugin-tech-insights                                  0.3.20
  @backstage/plugin-tech-radar                                     0.6.11
  @backstage/plugin-techdocs-backend                               1.9.1
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.3
  @backstage/plugin-techdocs-node                                  1.11.0
  @backstage/plugin-techdocs-react                                 1.1.14
  @backstage/plugin-techdocs                                       1.9.2
  @backstage/plugin-user-settings                                  0.7.14
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.4.6
  @backstage/theme                                                 0.4.4, 0.5.0
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
```

**Comment by Rugvip** [comment]
I'm not able to reproduce. In a newly created project using the launch configuration you posted I'm able to both add breakpoints in the main `.ts` source files in the project, as well as click through "Go to Source Definition" into dependencies and add breakpoints there as well.

**Comment by sonikro** [comment]
I'm having exactly the same problem. Weird thing is, I've been working with backstage for a long time now and it only started happening after I upgraded from 1.18 to 1.20. 

I'm also on DevContainers (on Windows with WSL2)

```
OS:   Linux 5.10.16.3-microsoft-standard-WSL2 - linux/x64
node: v18.17.0
yarn: 1.22.19
cli:  0.24.0 (installed)
backstage:  1.20.3

Dependencies:
  @backstage/app-defaults                                          1.4.5
  @backstage/backend-app-api                                       0.5.10-next.1, 0.5.8, 0.5.9
  @backstage/backend-common                                        0.19.9, 0.20.0, 0.20.1-next.1
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.1.0, 0.1.1, 0.1.2-next.1
  @backstage/backend-plugin-api                                    0.6.8, 0.6.9-next.1
  @backstage/backend-tasks                                         0.5.12, 0.5.13, 0.5.14-next.1
  @backstage/backend-test-utils                                    0.2.9
  @backstage/catalog-client                                        1.5.1, 1.5.2-next.0
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.1.5, 0.2.0, 0.2.1, 0.2.2-next.0
  @backstage/cli                                                   0.22.14, 0.24.0
  @backstage/config-loader                                         1.5.3, 1.6.0, 1.6.1-next.0
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.1, 1.11.2
  @backstage/core-compat-api                                       0.1.0
  @backstage/core-components                                       0.13.9
  @backstage/core-plugin-api                                       1.8.1
  @backstage/dev-utils                                             1.0.24
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.3
  @backstage/frontend-plugin-api                                   0.3.0, 0.4.0
  @backstage/integration-aws-node                                  0.1.8
  @backstage/integration-react                                     1.1.22
  @backstage/integration                                           1.7.2, 1.8.0
  @backstage/plugin-api-docs                                       0.10.1
  @backstage/plugin-app-backend                                    0.3.56
  @backstage/plugin-app-node                                       0.1.8
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.1
  @backstage/plugin-auth-backend-module-github-provider            0.1.4
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.4
  @backstage/plugin-auth-backend-module-google-provider            0.1.4
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.4
  @backstage/plugin-auth-backend                                   0.20.0
  @backstage/plugin-auth-node                                      0.4.2, 0.4.3-next.1
  @backstage/plugin-catalog-backend-module-github                  0.4.7-next.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.5
  @backstage/plugin-catalog-backend                                1.15.0, 1.16.1-next.1
  @backstage/plugin-catalog-common                                 1.0.19
  @backstage/plugin-catalog-graph                                  0.3.2
  @backstage/plugin-catalog-import                                 0.10.4
  @backstage/plugin-catalog-node                                   1.5.0, 1.6.0, 1.6.1-next.1
  @backstage/plugin-catalog-react                                  1.9.2
  @backstage/plugin-catalog                                        1.15.1
  @backstage/plugin-events-backend-module-github                   0.1.18
  @backstage/plugin-events-backend                                 0.2.17
  @backstage/plugin-events-node                                    0.2.17, 0.2.18-next.1
  @backstage/plugin-github-actions                                 0.6.9
  @backstage/plugin-github-issues                                  0.2.17
  @backstage/plugin-github-pull-requests-board                     0.1.22
  @backstage/plugin-home-react                                     0.1.5
  @backstage/plugin-home                                           0.5.11
  @backstage/plugin-kubernetes-backend                             0.13.1
  @backstage/plugin-kubernetes-common                              0.7.1, 0.7.2
  @backstage/plugin-kubernetes-node                                0.1.1
  @backstage/plugin-kubernetes-react                               0.2.0
  @backstage/plugin-kubernetes                                     0.11.3
  @backstage/plugin-org                                            0.6.18
  @backstage/plugin-permission-backend                             0.5.31
  @backstage/plugin-permission-common                              0.7.11
  @backstage/plugin-permission-node                                0.7.19, 0.7.20-next.1
  @backstage/plugin-permission-react                               0.4.18
  @backstage/plugin-proxy-backend                                  0.4.6
  @backstage/plugin-scaffolder-backend-module-azure                0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.1.0
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.0
  @backstage/plugin-scaffolder-backend-module-github               0.1.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.11
  @backstage/plugin-scaffolder-backend                             1.19.2
  @backstage/plugin-scaffolder-common                              1.4.3, 1.4.4
  @backstage/plugin-scaffolder-node                                0.2.9
  @backstage/plugin-scaffolder-react                               1.6.1
  @backstage/plugin-scaffolder                                     1.16.1
  @backstage/plugin-search-backend-module-catalog                  0.1.12, 0.1.13-next.1
  @backstage/plugin-search-backend-module-elasticsearch            1.3.11
  @backstage/plugin-search-backend-module-techdocs                 0.1.12
  @backstage/plugin-search-backend-node                            1.2.12, 1.2.13-next.1
  @backstage/plugin-search-backend                                 1.4.8
  @backstage/plugin-search-common                                  1.2.8, 1.2.9
  @backstage/plugin-search-react                                   1.7.4
  @backstage/plugin-search                                         1.4.4
  @backstage/plugin-tech-insights-backend-module-jsonfc            0.1.40
  @backstage/plugin-tech-insights-backend                          0.5.22
  @backstage/plugin-tech-insights-common                           0.2.12
  @backstage/plugin-tech-insights-node                             0.4.14
  @backstage/plugin-tech-insights                                  0.3.20
  @backstage/plugin-tech-radar                                     0.6.11
  @backstage/plugin-techdocs-backend                               1.9.1
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.3
  @backstage/plugin-techdocs-node                                  1.11.0, 1.10.0
  @backstage/plugin-techdocs-react                                 1.1.14
  @backstage/plugin-techdocs                                       1.9.2
  @backstage/plugin-user-settings                                  0.7.13
  @backstage/release-manifests                                     0.0.10, 0.0.11
  @backstage/test-utils                                            1.4.6
  @backstage/theme                                                 0.4.4, 0.5.0
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
```

I'll keep this issue updated if I find any solution.

**Comment by sonikro** [comment]
I tested my theory that it stopped working after the last upgrade, so I rolled back the **@backstage/cli** package to the version I was using before (0.22.13) and immediately all of the breakpoints are now binding:

See my new info:

```
OS:   Linux 5.10.16.3-microsoft-standard-WSL2 - linux/x64
node: v18.17.0
yarn: 1.22.19
cli:  0.22.13 (installed)
backstage:  1.20.3

Dependencies:
  @backstage/app-defaults                                          1.4.5
  @backstage/backend-app-api                                       0.5.10-next.1, 0.5.8, 0.5.9
  @backstage/backend-common                                        0.19.9, 0.20.0, 0.20.1-next.1
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.1.0, 0.1.1, 0.1.2-next.1
  @backstage/backend-plugin-api                                    0.6.8, 0.6.9-next.1
  @backstage/backend-tasks                                         0.5.12, 0.5.13, 0.5.14-next.1
  @backstage/backend-test-utils                                    0.2.9
  @backstage/catalog-client                                        1.5.1, 1.5.2-next.0
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.1.5, 0.2.0, 0.2.1, 0.2.2-next.0
  @backstage/cli                                                   0.22.13
  @backstage/config-loader                                         1.5.3, 1.6.0, 1.6.1-next.0
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.1, 1.11.2
  @backstage/core-compat-api                                       0.1.0
  @backstage/core-components                                       0.13.9
  @backstage/core-plugin-api                                       1.8.1
  @backstage/dev-utils                                             1.0.24
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.3
  @backstage/frontend-plugin-api                                   0.3.0, 0.4.0
  @backstage/integration-aws-node                                  0.1.8
  @backstage/integration-react                                     1.1.22
  @backstage/integration                                           1.7.2, 1.8.0
  @backstage/plugin-api-docs                                       0.10.1
  @backstage/plugin-app-backend                                    0.3.56
  @backstage/plugin-app-node                                       0.1.8
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.1
  @backstage/plugin-auth-backend-module-github-provider            0.1.4
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.4
  @backstage/plugin-auth-backend-module-google-provider            0.1.4
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.4
  @backstage/plugin-auth-backend                                   0.20.0
  @backstage/plugin-auth-node                                      0.4.2, 0.4.3-next.1
  @backstage/plugin-catalog-backend-module-github                  0.4.7-next.1
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.5
  @backstage/plugin-catalog-backend                                1.15.0, 1.16.1-next.1
  @backstage/plugin-catalog-common                                 1.0.19
  @backstage/plugin-catalog-graph                                  0.3.2
  @backstage/plugin-catalog-import                                 0.10.4
  @backstage/plugin-catalog-node                                   1.5.0, 1.6.0, 1.6.1-next.1
  @backstage/plugin-catalog-react                                  1.9.2
  @backstage/plugin-catalog                                        1.15.1
  @backstage/plugin-events-backend-module-github                   0.1.18
  @backstage/plugin-events-backend                                 0.2.17
  @backstage/plugin-events-node                                    0.2.17, 0.2.18-next.1
  @backstage/plugin-github-actions                                 0.6.9
  @backstage/plugin-github-issues                                  0.2.17
  @backstage/plugin-github-pull-requests-board                     0.1.22
  @backstage/plugin-home-react                                     0.1.5
  @backstage/plugin-home                                           0.5.11
  @backstage/plugin-kubernetes-backend                             0.13.1
  @backstage/plugin-kubernetes-common                              0.7.1, 0.7.2
  @backstage/plugin-kubernetes-node                                0.1.1
  @backstage/plugin-kubernetes-react                               0.2.0
  @backstage/plugin-kubernetes                                     0.11.3
  @backstage/plugin-org                                            0.6.18
  @backstage/plugin-permission-backend                             0.5.31
  @backstage/plugin-permission-common                              0.7.11
  @backstage/plugin-permission-node                                0.7.19, 0.7.20-next.1
  @backstage/plugin-permission-react                               0.4.18
  @backstage/plugin-proxy-backend                                  0.4.6
  @backstage/plugin-scaffolder-backend-module-azure                0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.1.0
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.0
  @backstage/plugin-scaffolder-backend-module-github               0.1.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.11
  @backstage/plugin-scaffolder-backend                             1.19.2
  @backstage/plugin-scaffolder-common                              1.4.3, 1.4.4
  @backstage/plugin-scaffolder-node                                0.2.9
  @backstage/plugin-scaffolder-react                               1.6.1
  @backstage/plugin-scaffolder                                     1.16.1
  @backstage/plugin-search-backend-module-catalog                  0.1.12, 0.1.13-next.1
  @backstage/plugin-search-backend-module-elasticsearch            1.3.11
  @backstage/plugin-search-backend-module-techdocs                 0.1.12
  @backstage/plugin-search-backend-node                            1.2.12, 1.2.13-next.1
  @backstage/plugin-search-backend                                 1.4.8
  @backstage/plugin-search-common                                  1.2.8, 1.2.9
  @backstage/plugin-search-react                                   1.7.4
  @backstage/plugin-search                                         1.4.4
  @backstage/plugin-tech-insights-backend-module-jsonfc            0.1.40
  @backstage/plugin-tech-insights-backend                          0.5.22
  @backstage/plugin-tech-insights-common                           0.2.12
  @backstage/plugin-tech-insights-node                             0.4.14
  @backstage/plugin-tech-insights                                  0.3.20
  @backstage/plugin-tech-radar                                     0.6.11
  @backstage/plugin-techdocs-backend                               1.9.1
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.3
  @backstage/plugin-techdocs-node                                  1.11.0, 1.10.0
  @backstage/plugin-techdocs-react                                 1.1.14
  @backstage/plugin-techdocs                                       1.9.2
  @backstage/plugin-user-settings                                  0.7.13
  @backstage/release-manifests                                     0.0.10
  @backstage/test-utils                                            1.4.6
  @backstage/theme                                                 0.4.4, 0.5.0
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7
```

The only difference is the backstage-cli version going from **0.24.0** back to **0.22.13**.

Might be worth investigating what changes were made between these versions and how they might be impacting debugging on vscode

**Comment by Ben-Ackerman** [comment]
I work on the same team as @sonikro.   Looking into recent changes to the backstage-cli since version 0.22.13 I noticed this commit https://github.com/backstage/backstage/commit/7077dbf1317bdf2c7e385f00bf7ae954ae6c41cb which changes the default start command to the EXPERIMENTAL_BACKEND_START command.

We tested 0.24.0 with LEGACY_BACKEND_START as a workaround and it worked for us.

This seems to be an issue with the transpile process in the new default command.
